packages/table/src/common/internal/scrollUtils.ts

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ * Licensed under the BSD-3 License as modified (the “License”); you may obtain a copy
+ * of the license at https://github.com/palantir/blueprint/blob/master/LICENSE
+ * and https://github.com/palantir/blueprint/blob/master/PATENTS
+ */
+
+import { IRegion, RegionCardinality, Regions } from "../../regions";
+
+// Public
+// ======
+
+export function getScrollPositionForRegion(
+    region: IRegion,
+    currScrollLeft: number,
+    currScrollTop: number,
+    getLeftOffset: (columnIndex: number) => number,
+    getTopOffset: (rowIndex: number) => number,
+    numFrozenRows: number = 0,
+    numFrozenColumns: number = 0,
+) {
+    const cardinality = Regions.getRegionCardinality(region);
+
+    let nextTop = currScrollTop;
+    let nextLeft = currScrollLeft;
+
+    // if these were max-frozen-index values, we would have added 1 before passing to the get*Offset
+    // functions, but the counts are already 1-indexed, so we can just pass those.
+    const frozenColumnsCumulativeWidth = getLeftOffset(numFrozenColumns);
+    const frozenRowsCumulativeHeight = getTopOffset(numFrozenRows);
+
+    if (cardinality === RegionCardinality.CELLS) {
+        // scroll to the top-left corner of the block of cells
+        const topOffset = getTopOffset(region.rows[0]);
+        const leftOffset = getLeftOffset(region.cols[0]);
+        nextTop = getAdjustedScrollPosition(topOffset, frozenRowsCumulativeHeight);
+        nextLeft = getAdjustedScrollPosition(leftOffset, frozenColumnsCumulativeWidth);
+    } else if (cardinality === RegionCardinality.FULL_ROWS) {
+        // scroll to the top of the row block
+        const topOffset = getTopOffset(region.rows[0]);
+        nextTop = getAdjustedScrollPosition(topOffset, frozenRowsCumulativeHeight);
+    } else if (cardinality === RegionCardinality.FULL_COLUMNS) {
+        // scroll to the left side of the column block
+        const leftOffset = getLeftOffset(region.cols[0]);
+        nextLeft = getAdjustedScrollPosition(leftOffset, frozenColumnsCumulativeWidth);
+    } else {
```

Comment:
this might make more sense as a `switch`? cuz that's exactly what it is, on enum values.

Reply:
I like to avoid switches unless I can `return` immediately within each case; otherwise I have an extra `break;` line at the end of every case, which I find bloat-y and unsightly.

Reply:
Also, TIL if you redeclare variables with the same name within different `switch` cases (e.g. `topOffset` and `leftOffset`), you get this compilation error: `[ts] Cannot redeclare block-scoped variable 'leftOffset'.`

Reply:
you can always put your switch-case in a block

```
case 1: {
   const foo = 1;
   break;
}
case 2: {
   const foo = 2;
   break;
}

Reply:
`break`s are annoying yes, but the benefit of compile-time warnings shouldn't be discarded

Reply:
@themadcreator nice, will change